internal/lsp/semantic_tokens.go

```diff
@@ -172,6 +177,64 @@ var builtins = map[string]SemanticTokenType{
 	"verdicttype":          Type,
 }
 
+type Range struct {
+	Begin loc.Pos
+	End   loc.Pos
+}
+
+type SemTokSeqItem struct {
+	Data []uint32
+	Idx  int
+}
+
+func LastTokenLine(d *[]uint32) uint32 {
+	var sum uint32 = 0
+	for i := 0; i < len(*d); i += 5 {
+		sum += (*d)[i]
+	}
+	return sum
+}
+
+func SemanticTokenReassambly(seqItems []SemTokSeqItem) []uint32 {
+	var lTokL uint32 = 0
+	d := make([]uint32, 0, len(seqItems[0].Data))
+	for i, item := range seqItems {
+		if i == 0 {
+			d = append(d, item.Data...)
+			lTokL = LastTokenLine(&item.Data)
+		} else {
+			baseIdx := len(d)
+			d = append(d, item.Data...)
+			d[baseIdx] = d[baseIdx] - lTokL //delta-line
+			lTokL = LastTokenLine(&item.Data)
+		}
+	}
+	return d
+}
+
+func CalculateEqualLineRanges(tree *ttcn3.Tree, b loc.Pos, e loc.Pos, splitAfterLines uint, nrOfRanges uint) []Range {
+	res := make([]Range, 0, 2)
+	begin := tree.Position(b)
+	end := tree.Position(e)
+	r := end.Line - begin.Line
+	if r > int(splitAfterLines) {
+		for i := 0; i < int(nrOfRanges); i++ {
+			nextEndL := begin.Line + r/int(nrOfRanges)
+			if i == int(nrOfRanges-1) {
+				res = append(res, Range{tree.Pos(begin.Line, begin.Column), e})
+			} else {
+				res = append(res, Range{tree.Pos(begin.Line, begin.Column), tree.Pos(nextEndL, 1) - 1})
+			}
+			begin.Column = 1
+			begin.Line += r / int(nrOfRanges)
+		}
+	} else {
```

Comment:
It's considered good practice to do without else clause by early return. For example:
```go
r := end.Line - begin.Line
if r <= splitAfterlines {
       return Range{b, e}
}
for i := 0; i < nrOfRangees; i++ {
```

But in this case you could move the condition to the caller even?